IdentityCore/src/oauth2/appmetadata/MSIDGeneralCacheItemType.m

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDGeneralCacheItemType.h"
+
+@implementation MSIDGeneralCacheItemTypeHelpers
+
++ (NSString *)generalTypeAsString:(MSIDGeneralCacheItemType)type
+{
+    switch (type)
+    {
+        case MSIDAppMetadataType:
+            return MSID_APP_METADATA_CACHE_TYPE;
+            
+        default:
+            return nil;
+    }
+}
+
+static NSDictionary *sGeneralTypes = nil;
+
++ (MSIDGeneralCacheItemType)generalTypeFromString:(NSString *)type
```

Comment:
can you please do case insensitive match for the type string?

Reply:
This type is passed to a dictionary . Case insensitive compare will create a different hash in the dictionary I think. We do the same thing in MSIDCredentialType right?

Reply:
Check credentialtype implementation in dev, it lowercases the type before comparing.